test/unit/test_modules.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import os
+import subprocess
+import sys
+
+from mock import call, mock_open, patch
+import pytest
+from six import PY2
+
+import sagemaker_containers as smc
+
+builtins_open = '__builtin__.open' if PY2 else 'builtins.open'
+
+
+@patch('boto3.resource', autospec=True)
+@pytest.mark.parametrize('url,bucket_name,key,dst', [
+    ('S3://my-bucket/path/to/my-file', 'my-bucket', 'path/to/my-file', '/tmp/my-file'),
+    ('s3://my-bucket/my-file', 'my-bucket', 'my-file', '/tmp/my-file')
+])
+def test_s3_download(resource, url, bucket_name, key, dst):
+    smc.modules.s3_download(url, dst)
+
+    chain = call('s3').Bucket(bucket_name).download_file(key, dst)
+    assert resource.mock_calls == chain.call_list()
```

Comment:
Does this test correctly verify the arguments passed to .Bucket and download_file? e.g. if you change the parameters in the call to create "chain", does the test fail?

Reply:
It does ! https://docs.python.org/3/library/unittest.mock-examples.html#setting-return-values-and-attributes